adapter/internal/operator/controllers/dp/api_controller.go

```diff
@@ -64,15 +66,21 @@ func NewAPIController(mgr manager.Manager, operatorDataStore *synchronizer.Opera
 		return err
 	}
 
-	if err := c.Watch(&source.Kind{Type: &dpv1alpha1.API{}}, &handler.EnqueueRequestForObject{}); err != nil {
+	conf := config.ReadConfigs()
+	predicates := []predicate.Predicate{predicate.NewPredicateFuncs(utils.FilterByNamespaces(conf.Adapter.Operator.Namespaces))}
```

Comment:
It is handled to watch for all the namespaces in `FilterByNamespaces` function. I didn't add pod namespace scenario here since it will break the default case of watching all namespaces. So, user needs to add it in the namespaces config for the scenario you described.

Reply:
Ack. That's correct.